### PR TITLE
Feature/remove session store

### LINKED ISSUE #2208: Remove ET back-end components
As a part of https://github.com/kedro-org/kedro-viz/issues/2202, this ticket is to remove the BE components. This includes GraphQL Strawberry, the SQL session store, and associated tests.

- [ ]  Remove graphql folder from api
- [ ]  Remove session store
- [ ]  Remove “/et” from apps.py (line 84)

### PR DESCRIPTION
**Pull request by Huongg**
## Description
Fixes #2208
This PR is part of the effort to remove experiment tracking by removing the session store from the back-end.

Main Changes:

- [x] Completely removed database.py.
- [x] Completely removed sqlite_store.py.
- [x] Removed session_store references from:
  - data_loader.py
  - server.py
- [x] Deleted all tests related to the session store.

QA Notes:
Please pull the changes locally and check that the app runs without errors, 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: demo-project/src/demo_project/settings.py [hunk 1]
```diff
@@ -2,25 +2,6 @@
...
-from kedro_viz.integrations.kedro.sqlite_store import SQLiteStore
-
-SESSION_STORE_CLASS = SQLiteStore
-SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data")}
-
-# Setup for collaborative experiment tracking.
-# SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data"),
-#                       "remote_path": "s3://{path-to-session_store}" }
-
-# Define custom context class. Defaults to `KedroContext`
```
Comment (ravi-kumar-pilla) [inline demo-project/src/demo_project/settings.py:18]: I think you can keep the code from line 18-23 as is, as this is not related to ET

File: package/kedro_viz/database.py [hunk 2]
```diff
@@ -21,21 +21,3 @@ def configure_wal_for_azure(engine):
     if is_azure_ml:
         with engine.connect() as conn:
             conn.execute(text("PRAGMA journal_mode=WAL;"))
-
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/database.py:24]: This file is for handling SQLite which is not needed anymore, so I guess we can delete the file entirely

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -210,14 +210,6 @@ make run PROJECT_PATH=<path-to-your-test-project>/new-kedro-project
...
 > **Note**: Once the backend development server is launched at port 4142, the local app will always pull data from that server. To prevent this, you can comment out the proxy setting in `package.json` and restart the dev server at port 4141.
 
-#### Launch the development server with the `SQLiteSessionStore`
-
-Kedro-Viz provides a `SQLiteSessionStore` that users can use in their project to enable experiment tracking functionality. If you want to use this session store with the development server, make sure you don't use a relative path when specifying the store's location in `settings.py`. For example, `demo-project` specifies the local `data` directory within a project as the session store's location as follows:
-
-```python
-from kedro_viz.integrations.kedro.sqlite_store import SQLiteStore
-SESSION_STORE_ARGS = {"path": str(Path(__file__).parents[2] / "data")}
-```
```
Comment (ravi-kumar-pilla) [inline CONTRIBUTING.md:220]: We need to make changes to all the starter templates (may be a note saying kedro-viz 10.1.0 does not support ET) as a follow up - https://github.com/kedro-org/kedro-starters/blob/main/spaceflights-pandas-viz/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/settings.py

**Comment by ravi-kumar-pilla** [comment]
Hi @Huongg ,

I know this PR is related to removing only Session store. I had a brief look and it looks fine for me. In general, I prepared a checklist as reference (you may add to this, but overall this would suffice) -

- [ ] Remove graphql folder from api
- [ ] Remove graphql_router from apps.py (line 66)
- [ ] Remove “/et” from apps.py (line 84)
- [ ] Remove SQLite Store from integrations/kedro/sqlite_store (This might also require changes to kedro starters as we say in the docs for ET, configure SQLiteStore in settings.py)
- [ ] Remove docs related to ET

I hope this helps. I will also have a look at this PR once it is ready for review. Thank you

**Comment by Huongg** [comment]
Thanks, @ravi-kumar-pilla, for the helpful guidance! I’ve implemented your suggestion to update the code related to removing session_store. The changes related to GraphQL and the router will be addressed in a separate PR to keep this one more focused and easier to review. Please take a look and let me know your thoughts.

Currently, the test coverage isn’t at 100% because I’ve deleted all the graphql_tests. Since the GraphQL-related code will be removed soon, I’m wondering if it's okay to ignore this in the PR for now. Here's the link to the error for reference: https://github.com/kedro-org/kedro-viz/actions/runs/12160128272/job/33911739972?pr=2219

File: CONTRIBUTING.md (hunk 3 above)
Comment (Huongg) [reply CONTRIBUTING.md]: hey, @merelcht has already raised this concern, and I’ll create a ticket for it. For now, we’ve added [a deprecation warning for Kedro datasets](https://github.com/kedro-org/kedro-plugins/issues/954) to alert users about this change coming in the next release.

**Comment by tynandebold** [comment]
Is the plan to remove all of the graphql code, or only that which relates to experiment tracking?

I'm in favor of removing it entirely. There's one graphql request that's used on the flowchart page though, and that's the `getVersion` one. That could definitely be moved to a REST endpoint. Bringing up here so we don't forget about it.

**Comment by Huongg** [comment]
> Is the plan to remove all of the graphql code, or only that which relates to experiment tracking?
> 
> I'm in favor of removing it entirely. There's one graphql request that's used on the flowchart page though, and that's the `getVersion` one. That could definitely be moved to a REST endpoint. Bringing up here so we don't forget about it.

hey so I'm thinking we're removing the graphql code related to ET, and I've opened a new issue separately to update the `getVersion` [here](https://github.com/kedro-org/kedro-viz/issues/2221)

**Review by jitu5 — APPROVED** [review]
Thanks @Huongg

**Review by SajidAlamQB — APPROVED** [review]
Awesome work @Huongg, I've looked through most of the changes and it LGTM!
